polling/src/main/java/io/atleon/polling/reactive/Poller.java

```diff
@@ -0,0 +1,34 @@
+package io.atleon.polling.reactive;
+
+import java.util.Collection;
+
+/**
+ * The Poller class is what manages the interaction with the external resource to be polled. The Poller
+ * should be implemented by the user and supplied to {@link PollingReceiver} when instantiated. The supplied
+ * implementation will need to have an internal mechanism to track offsets into the external resource as this
+ * poll() method has no parameters and is only expecting to receive the next batch of events for the stream.
+ * @param <K> The type of the items returned by the external resource (Must implement {@link PollingRecord}.
+ * @param <V> The data type of the data needed for Acking and Nacking.
+ */
+public interface Poller<K extends PollingRecord<V>, V> {
```

Comment:
An oddity with the naming here just a occurred to me: This is called "Poller", but I'm not sure it's actually the one doing the polling; It's rather encapsulating the behavior of something that can be polled. I wonder if `Pollable` would be a better name?